### PR TITLE
[BEAM-9936] Create SDK harness containers with Python 3.8

### PR DESCRIPTION
**Pull request by kamilwu**
List of changes:
* Base image is now based on Debian Buster
* new gradle project `:sdks:python:container:py38:docker`
* a license for `tensorboard-plugin-wit`, a new dependency introduced by Tensorboard 2.20

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by kamilwu** [comment]
Run Python Dataflow ValidatesContainer

**Comment by kamilwu** [comment]
Run Python Dataflow ValidatesContainer

**Comment by kamilwu** [comment]
Run PythonLint PreCommit

**Comment by kamilwu** [comment]
Run Python Dataflow ValidatesContainer

**Comment by kamilwu** [comment]
Run Java PreCommit

**Comment by kamilwu** [comment]
Run Python PreCommit

**Comment by kamilwu** [comment]
Run Java PreCommit

**Comment by kamilwu** [comment]
R: @Hannah-Jiang 
cc: @tvalentyn

File: sdks/python/container/run_validatescontainer.sh [hunk 1]
```diff
@@ -69,8 +71,13 @@ elif [[ $1 == "python37" ]]; then
   CONTAINER_PROJECT="sdks:python:container:py37"  # Use this to build container by Gradle.
   GRADLE_PY3_FLAG="-Ppython3"   # Use this in Gradle command.
   PY_INTERPRETER="python3.7"    # Use this in virtualenv command.
+elif [[ $1 == "python38" ]]; then
+  IMAGE_NAME="${IMAGE_PREFIX}python3.8_sdk"    # Use this to create CONTAINER_IMAGE variable.
+  CONTAINER_PROJECT="sdks:python:container:py38"  # Use this to build container by Gradle.
+  GRADLE_PY3_FLAG="-Ppython3"   # Use this in Gradle command.
```
[the remark this answers is not in the document]
Comment (kamilwu) [reply sdks/python/container/run_validatescontainer.sh]: Indeed there's no difference after removal. Looks like it was a leftover junk.

**Comment by kamilwu** [comment]
Run Python PreCommit

**Comment by kamilwu** [comment]
Run Java PreCommit

**Comment by kamilwu** [comment]
Run Java PreCommit

**Comment by tvalentyn** [comment]
@ibzib Do you know if we will need to update release instructions to make sure we build and release Python 3.8 containers for future releases? (It may be done in a future PR).

**Comment by tvalentyn** [comment]
Run Python Dataflow ValidatesContainer

**Comment by ibzib** [comment]
> @ibzib Do you know if we will need to update release instructions to make sure we build and release Python 3.8 containers for future releases? (It may be done in a future PR).

All we need is to add 3.8 to these two scripts:
https://github.com/apache/beam/blob/836d0ad5b190502ac2086b9c44f4a7e8f4f9625d/release/src/main/scripts/build_release_candidate.sh#L51
https://github.com/apache/beam/blob/836d0ad5b190502ac2086b9c44f4a7e8f4f9625d/release/src/main/scripts/publish_docker_images.sh#L30

**Comment by ibzib** [comment]
> > @ibzib Do you know if we will need to update release instructions to make sure we build and release Python 3.8 containers for future releases? (It may be done in a future PR).
> 
> All we need is to add 3.8 to these two scripts:
> https://github.com/apache/beam/blob/836d0ad5b190502ac2086b9c44f4a7e8f4f9625d/release/src/main/scripts/build_release_candidate.sh#L51
> 
> 
> https://github.com/apache/beam/blob/836d0ad5b190502ac2086b9c44f4a7e8f4f9625d/release/src/main/scripts/publish_docker_images.sh#L30

I also filed a ticket with INFRA to add the new repo on Docker hub, since no one on the Beam project has those privileges: https://issues.apache.org/jira/browse/INFRA-20288

**Comment by tvalentyn** [comment]
Awesome, thank you!
I also opened BEAM-10046 to make sure Python 3.8 is reflected in the Beam release.

**Comment by kamilwu** [comment]
Run Python Dataflow ValidatesContainer

**Comment by kamilwu** [comment]
Run Python2_PVR_Flink PreCommit

**Comment by kamilwu** [comment]
Run SQL_Java11 PreCommit

**Comment by tvalentyn** [comment]
LGTM, thanks, @kamilwu !

**Comment by ibzib** [comment]
I think this PR might have caused BEAM-10316, PTAL

**Pull request merged**